test/config-luatest/validate_names_test.lua

```diff
@@ -0,0 +1,167 @@
+local t = require('luatest')
+local server = require('test.luatest_helpers.server')
+local configdata = require('internal.config.configdata')
+local cluster_config = require('internal.config.cluster_config')
+local replica_set = require('luatest.replica_set')
+local uuid = require('uuid')
+
+local g_names = t.group('config_recovery_with_names_set')
+
+--
+-- before_all trigger starts cluster in order to generate snapshot
+-- files, which will be used to test, how confidata works.
+--
+local function cluster_initialize(g, with_names)
+    local replicaset_uuid = uuid.str()
+    local instance_1_uuid = uuid.str()
+    local instance_2_uuid = uuid.str()
+
+    g.replica_set = replica_set:new({})
+    local box_cfg = {
+        replicaset_name = with_names and 'replicaset-001' or nil,
+        replicaset_uuid = replicaset_uuid,
+        replication = {
+            server.build_listen_uri('instance-001', g.replica_set.id),
+            server.build_listen_uri('instance-002', g.replica_set.id),
+        },
+    }
+
+    box_cfg.instance_name = with_names and 'instance-001' or nil
+    box_cfg.instance_uuid = instance_1_uuid
+    local instance_1 = g.replica_set:build_and_add_server(
+        {alias = 'instance-001', box_cfg = box_cfg})
+
+    box_cfg.instance_name = with_names and 'instance-002' or nil
+    box_cfg.instance_uuid = instance_2_uuid
+    box_cfg.read_only = true
+    local instance_2 = g.replica_set:build_and_add_server(
+        {alias = 'instance-002', box_cfg = box_cfg})
+
+    g.replica_set:start()
+    g.replica_set:wait_for_fullmesh()
+    for _, replica in ipairs(g.replica_set.servers) do
+        replica:exec(function()
+            box.snapshot()
+        end)
+    end
+
+    g.workdir_1 = instance_1.workdir
+    g.workdir_2 = instance_2.workdir
+
+    g.config = {
+        credentials = {
+            users = {
+                guest = {
+                    roles = {'super'},
+                },
+            },
+        },
+
+        snapshot = {
+            dir = g.workdir_1,
+        },
+
+        iproto = {
+            listen = 'unix/:./{{ instance_name }}.iproto',
+        },
+
+        groups = {
+            ['group-001'] = {
+                replicasets = {
+                    ['replicaset-001'] = {
+                        database = { replicaset_uuid = replicaset_uuid },
+                        instances = {
+                            ['instance-001'] = {
+                                database = {instance_uuid = instance_1_uuid},
+                            },
+                            ['instance-002'] = {
+                                database = {instance_uuid = instance_2_uuid},
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+    -- Simulate startup procedure for configdata.
+    box.cfg = function() end
```

Comment:
Nit: I wouldn't do it inside a tarantool process that executes luatest logic. I think it is better to do in a separate tarantool instance.

Reply:
Sure, done